### PR TITLE
USHIFT-285: Add MicroShift version getter to framework

### PR DESCRIPTION
**Pull request by pacevedom**
Add a function to determine if the cluster is a MicroShift instance.

### CODE REVIEW

File: test/extended/util/framework.go
Comment (dhellmann) [inline test/extended/util/framework.go:2147]: The version string is going to be the full display version, so you could just return that instead of concatenating it with the major and minor versions.

Comment (pacevedom) [reply test/extended/util/framework.go:2147]: Mind if we change it in a future commit? As of today the version comes from the git version. It is true though that the `microShiftVersion` is only there for caching and (maybe) future convenience to extract the concrete version to do something with it.
In this change the caller is not able to see the version, only if its microshift or not.

Comment (pacevedom) [reply test/extended/util/framework.go:2137]: Fixed it, saw it was including the full version when compiling.

**Comment by dhellmann** [comment]
The implementation looks good. Are those CI job failures related?

**Comment by pacevedom** [comment]
> The implementation looks good. Are those CI job failures related?

Not really, this function is not used yet. Anyway I tmeporarily removed the call from `GetControlPlaneTopology` to inspect if we might enable bad code paths in other tests.

File: test/extended/util/framework.go
Comment (dhellmann) [inline test/extended/util/framework.go:2120]: This function would be simpler if this test was reversed and it returned the cached value immediately.

**Review by dhellmann — COMMENTED** [review]
One nit that can be addressed later.

/approve

File: test/extended/util/framework.go
Comment (pacevedom) [reply test/extended/util/framework.go:2120]: I thought of doing the same, but I chose to follow the existing pattern here: https://github.com/openshift/origin/blob/master/test/extended/util/framework.go#L2002-L2012

There is a kind of early return but its only under some special conditions a few lines below: https://github.com/openshift/origin/blob/master/test/extended/util/framework.go#L2030-L2042

Should I keep the style?

**Comment by ggiguash** [comment]
/retitle USHIFT-285: Add MicroShift version getter to framework

File: test/extended/util/framework.go
Comment (dhellmann) [reply test/extended/util/framework.go:2120]: ok, let's keep this version

**Review by dhellmann — COMMENTED** [review]
/lgtm

File: test/extended/util/framework.go
Comment (aojea) [inline test/extended/util/framework.go:2117]: why do you need a mutex?

Comment (pacevedom) [reply test/extended/util/framework.go:2117]: Tests run in parallel and maybe we have concurrent calls to this function. It's not something critical as the value would get reconstructed from immutable values in the cluster, but it saves calls to the API (only when starting tests, though), plus it follows the current style of other getters in the package:
https://github.com/openshift/origin/blob/master/test/extended/util/framework.go#L1982
https://github.com/openshift/origin/blob/master/test/extended/util/framework.go#L2027

Comment (aojea) [reply test/extended/util/framework.go:2117]: just curious, :smile:

**Review by ingvagabund — CHANGES_REQUESTED** [review]

**Comment by deads2k** [comment]
/hold

let's see where this is needed instead of using existing external signals before merging.

File: test/extended/util/framework.go
Comment (pacevedom) [reply test/extended/util/framework.go:2121]: Done

**Comment by pacevedom** [comment]
> /hold
> 
> let's see where this is needed instead of using existing external signals before merging.

So it's these tests now:
```
"[sig-auth][Feature:PodSecurity] restricted-v2 SCC should mutate empty securityContext to match restricted PSa profile [Suite:openshift/conformance/parallel]"
"[sig-auth][Feature:OpenShiftAuthorization] The default cluster RBAC policy should have correct RBAC rules [Suite:openshift/conformance/parallel]"
"[sig-coreos] [Conformance] CoreOS bootimages TestBootimagesPresent [Suite:openshift/conformance/parallel/minimal]"
"[sig-devex][Feature:OpenShiftControllerManager] TestAutomaticCreationOfPullSecrets [apigroup:config.openshift.io] [Suite:openshift/conformance/parallel]"
"[sig-devex][Feature:OpenShiftControllerManager] TestDockercfgTokenDeletedController [Suite:openshift/conformance/parallel]"
"[sig-imageregistry][Feature:ImageInfo] Image info should display information about images [Skipped:Disconnected] [Suite:openshift/conformance/parallel]"
"[sig-network][Feature:Router][apigroup:route.openshift.io] The HAProxy router converges when multiple routers are writing conflicting status [Suite:openshift/conformance/parallel]"
"[sig-network][Feature:Router][apigroup:route.openshift.io] The HAProxy router converges when multiple routers are writing status [Suite:openshift/conformance/parallel]"
```

More to come as we create/modify specific tests for microshift, I would say.

**Comment by pacevedom** [comment]
Not needed anymore, as we look for dependencies in each test.

**Pull request closed without merging**
